### PR TITLE
Add last activity functions for pointing device

### PR DESCRIPTION
**Pull request by drashna**
## Description

Add activity timers for pointing device. 

## Types of Changes

- [x] Core
- [x] Enhancement/optimization

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/keyboard.c [hunk 1]
```diff
@@ -627,27 +638,40 @@ void keyboard_task(void) {
     }
 #endif
 
+#ifdef POINTING_DEVICE_ENABLE
+    pointing_device_task();
+    const bool pointing_device_changed = !(last_pointing_device_activity_elapsed());
+#endif
```
Comment (drashna) [inline quantum/keyboard.c:644]: Not sure this is the best way to handle this. 

but short of significantly changing pointing device code, or tracking the changed state outside of the keyboard task loop...

**Review by tzarc — APPROVED** [review]

File: quantum/keyboard.c (hunk 1 above)
Comment (drashna) [reply quantum/keyboard.c:644]: That was may main concern, as well. 

I thought about changing pointing device task and send to be bool, but that would be a much more significant and invasive change.

Comment (tzarc) [reply quantum/keyboard.c:644]: Phase 2, if this PR's implementation is found to be suboptimal?

Comment (drashna) [reply quantum/keyboard.c:644]: > Phase 2, if this PR's implementation is found to be suboptimal?

Rewrite the task and send functions.  Make send only handle sending the report (and zeroing out the vector movement), and let the task handle the send... 

You know what, screw it... doing that now.

Comment (drashna) [reply quantum/keyboard.c]: rewrote anyways.  This should handle setting the reports outside of the pointing device task, and still properly report activity.

**Review by daskygit — APPROVED** [review]

**Pull request merged**